test/remote/gateways/remote_decidir_plus_test.rb

```diff
@@ -0,0 +1,90 @@
+require 'test_helper'
+require 'securerandom'
+
+class RemoteDecidirPlusTest < Test::Unit::TestCase
+  def setup
+    @gateway = DecidirPlusGateway.new(fixtures(:decidir_plus))
+
+    @amount = 100
+    @credit_card = credit_card('4484590159923090')
+    @declined_card = credit_card('4000300011112220')
+    @options = {
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    assert response = @gateway.store(@credit_card)
+
+    response = @gateway.purchase(@amount, @credit_card, @options.merge(payment_id: response.authorization))
+    assert_success response
+    assert_equal 'approved', response.message
+  end
+
+  def test_failed_purchase
+    assert response = @gateway.store(@credit_card)
+
+    @options[:site_transaction_id] = SecureRandom.hex
```

Comment:
This can be removed since the gateway does this as part of the purchase method now